Fields with pointer types cannot be unmarshalled

At some point this broke, so it's no longer possible to unmarshal a type with a pointer field. For example:

```go
func TestPointerFields(t *testing.T) {
	runner := func(dest interface{}) func(*testing.T) {
		return func(t *testing.T) {
			env := Values{"Data_Value_Size": {"16"}}

			r := Reader{Source: env, Sep: "_"}
			err := r.Getenv(dest, "Data")
			if err != nil {
				t.Fatalf("unexpected error: %v", err)
			}
		}
	}

	t.Run("StructValue", runner(new(
		struct {
			Value struct {
				Size int
			}
		},
	)))

	t.Run("StructPointer", runner(new(
		struct {
			Value *struct {
				Size int
			}
		},
	)))
}
```

The first subtest, StructValue, succeeds. The second, StructPointer, fails. This is supported in the internal version of envi, so I introduced a new bug between v0 and v1 during the refactoring. Amazingly, none of the existing tests actually covers this.

---

Stack trace:

```
panic: reflect.Set: value of type *struct { Size int } is not assignable to type struct { Size int } [recovered]
        panic: reflect.Set: value of type *struct { Size int } is not assignable to type struct { Size int }

goroutine 237 [running]:
testing.tRunner.func1(0xc42011db30)
        /usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x1129fa0, 0xc42011e710)
        /usr/local/go/src/runtime/panic.go:505 +0x229
reflect.Value.assignTo(0x1125760, 0xc42000c0d8, 0x196, 0x115ed15, 0xb, 0x11385e0, 0x0, 0x1125701, 0x1125760, 0xc420111b30)
        /usr/local/go/src/reflect/value.go:2235 +0x427
reflect.Value.Set(0x11385e0, 0xc420111b30, 0x199, 0x1125760, 0xc42000c0d8, 0x196)
        /usr/local/go/src/reflect/value.go:1373 +0xa4
github.com/Kochava/envi.(*Reader).loadStructField(0xc420081f60, 0x1138660, 0xc42000c0d8, 0x199, 0x117c7c0, 0x1138660, 0x0, 0x115d9b0, 0x4, 0x1138660, ...)
        …/github.com/Kochava/envi/reader.go:438 +0x3ec
github.com/Kochava/envi.(*Reader).loadStruct(0xc420081f60, 0x1138660, 0xc42000c0d8, 0x199, 0x115d9b0, 0x4, 0x20, 0xf4356b620113dc60)
        …/github.com/Kochava/envi/reader.go:398 +0x12c
github.com/Kochava/envi.(*Reader).loadReflect(0xc420045f60, 0x1125820, 0xc42000c0d8, 0x0, 0x0, 0x115d9b0, 0x4, 0x0, 0x0)
        …/github.com/Kochava/envi/reader.go:217 +0x23a
github.com/Kochava/envi.(*Reader).load(0xc420045f60, 0x1125820, 0xc42000c0d8, 0x0, 0x0, 0x115d9b0, 0x4, 0x2b61c81951471eac, 0x100fdb8)
        …/github.com/Kochava/envi/reader.go:120 +0x1e9
github.com/Kochava/envi.(*Reader).Getenv(0xc420045f60, 0x1125820, 0xc42000c0d8, 0x115d9b0, 0x4, 0x117a840, 0xc420113740)
        …/github.com/Kochava/envi/reader.go:60 +0x2a3
github.com/Kochava/envi.TestPointerFields.func1.1(0xc42011db30)
        …/github.com/Kochava/envi/structptr_test.go:11 +0x16f
testing.tRunner(0xc42011db30, 0xc420113700)
        /usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:824 +0x2e0
exit status 2
FAIL    github.com/Kochava/envi 0.013s
```

--- a/issues_test.go
+++ b/issues_test.go
@@ -1,0 +1,51 @@
+package envi
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestPointerFields(t *testing.T) { // issue #2
+	runner := func(env Env, dest interface{}, want interface{}) func(*testing.T) {
+		return func(t *testing.T) {
+
+			r := Reader{Source: env, Sep: "_"}
+			err := r.Getenv(dest, "Data")
+			if err != nil && !IsNoValue(err) {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			if !reflect.DeepEqual(dest, want) {
+				t.Fatalf("got\t%#+ v\nwant\t%#+ v", dest, want)
+			}
+		}
+	}
+
+	type Data struct {
+		Size int
+	}
+
+	type Value struct {
+		Value Data
+	}
+
+	type Pointer struct {
+		Value *Data
+	}
+
+	t.Run("HasEnv", func(t *testing.T) {
+		env := Values{"Data_Value_Size": {"16"}}
+		t.Run("StructValue", runner(env, new(Value), &Value{Value: Data{Size: 16}}))
+		t.Run("StructPointer", runner(env, new(Pointer), &Pointer{Value: &Data{Size: 16}}))
+		t.Run("StructValueAssigned", runner(env, &Value{Value: Data{Size: 32}}, &Value{Value: Data{Size: 16}}))
+		t.Run("StructPointerAssigned", runner(env, &Pointer{Value: &Data{Size: 32}}, &Pointer{Value: &Data{Size: 16}}))
+	})
+
+	t.Run("WithoutEnv", func(t *testing.T) {
+		env := Values{}
+		t.Run("StructValue", runner(env, new(Value), &Value{Value: Data{}}))
+		t.Run("StructPointer", runner(env, new(Pointer), &Pointer{Value: nil}))
+		t.Run("StructValueAssigned", runner(env, &Value{Value: Data{Size: 32}}, &Value{Value: Data{Size: 32}}))
+		t.Run("StructPointerAssigned", runner(env, &Pointer{Value: &Data{Size: 32}}, &Pointer{Value: &Data{Size: 32}}))
+	})
+}

--- a/reader.go
+++ b/reader.go
@@ -435,7 +435,10 @@ func (r *Reader) loadStructField(out reflect.Value, typ reflect.Type, fieldIdx i
 		dst    = target.Interface()
 	)
 
-	target.Elem().Set(field)
+	// Only copy field's value to temporary storage if it's valid
+	if fi := reflect.Indirect(field); fi.IsValid() {
+		target.Elem().Set(fi)
+	}
 	if err = r.Getenv(dst, fname); err != nil && !IsNoValue(err) && !flags.quiet {
 		return isset, err
 	}

